addons/binding/org.openhab.binding.bosesoundtouch10/src/main/java/org/openhab/binding/bosesoundtouch/helper/XmlHelper.java

```diff
@@ -0,0 +1,192 @@
+        } catch (ParserConfigurationException | SAXException | IOException e) {
+            logger.debug("Exception on creating xml document");
+        }
+        return xmlDocument;
+    }
+
+    public synchronized Set<XmlResult> parseMessage(String message) {
+        Set<XmlResult> resultSet = new HashSet<XmlResult>();
+        if (message == null || message.isEmpty()) {
+            return resultSet;
+        }
+        String result = "";
+        try {
+            Document xmlDocument = createXmlDocument(message);
+            if (xmlDocument == null) {
+                return resultSet;
+            }
+            XPath xPath = XPathFactory.newInstance().newXPath();
+            for (String expression : SUPPORTED_MESSAGES_TYPE0) {
+                // XPATH does not work for this
+                if (message.contains("bassUpdated")) {
+                    resultSet.add(new XmlResult(expression, ""));
+                }
+            }
+            for (String expression : SUPPORTED_MESSAGES_TYPE1) {
+                result = xPath.compile(expression).evaluate(xmlDocument);
+                if (!result.isEmpty()) {
+                    resultSet.add(new XmlResult(expression, result));
+                }
+            }
+            NodeList nodeList;
+            Node node;
+            Element element;
+            for (String expression : SUPPORTED_MESSAGES_TYPE2) {
+                nodeList = (NodeList) xPath.compile(expression).evaluate(xmlDocument, XPathConstants.NODESET);
+                for (int i = 0; i < nodeList.getLength(); i++) {
+                    node = nodeList.item(i);
+                    if (node.getNodeType() == Node.ELEMENT_NODE) {
+                        element = (Element) node;
+                        if (!element.getAttribute(NODE_ATTRIBUTE_SOURCE).isEmpty()) {
+                            result = element.getAttribute(NODE_ATTRIBUTE_SOURCE);
+                            resultSet.add(new XmlResult(expression, result));
+                        }
+                        if (!element.getAttribute(NODE_ATTRIBUTE_ID).isEmpty()) {
+                            result = element.getAttribute(NODE_ATTRIBUTE_ID);
+                            resultSet.add(new XmlResult(expression, result));
+                        }
+                    }
+                }
+            }
+        } catch (XPathExpressionException e) {
+            logger.debug("Exception on XPath");
+        }
+        return resultSet;
+    }
+
+    public synchronized String parsePath(String expression, String message) {
+        String result = "";
+        if (message == null || message.isEmpty()) {
+            return "";
+        }
+        try {
+            Document xmlDocument = createXmlDocument(message);
+            XPath xPath = XPathFactory.newInstance().newXPath();
+            result = xPath.compile(expression).evaluate(xmlDocument);
+        } catch (XPathExpressionException e) {
+            logger.debug("Exception on XMLHelper");
```

Comment:
Do you really want to 'ignore' the exception like this, by debug logging and just continuing with the output ""?